extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/discovery/AstroDiscoveryLocationChangedTask.java

```diff
@@ -0,0 +1,23 @@
+package org.eclipse.smarthome.binding.astro.discovery;
+
+import org.eclipse.smarthome.core.library.types.PointType;
+
+public class AstroDiscoveryLocationChangedTask implements Runnable {
+
+    private AstroDiscoveryService astroDiscoveryService;
+    private PointType previousLocation;
+
+    public AstroDiscoveryLocationChangedTask(AstroDiscoveryService astroDiscoveryService) {
+        this.astroDiscoveryService = astroDiscoveryService;
+    }
+
+    @Override
+    public void run() {
+        PointType currentLocation = astroDiscoveryService.getLocationProvider().getLocation();
+        if (currentLocation != previousLocation) {
```

Comment:
Do you really want to compare the reference, regardless if another point type variable points to the same location? Shouldn't we use equals?